apps/omg_child_chain/test/omg_child_chain/integration/fee_server_test.exs

```diff
@@ -0,0 +1,155 @@
+# Copyright 2019 OmiseGO Pte Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+defmodule OMG.ChildChain.Integration.FeeServer do
+  @moduledoc """
+  Tests a simple happy path of all the pieces working together
+  """
+
+  use ExUnitFixtures
+  use ExUnit.Case, async: false
+  import ExUnit.CaptureLog
+
+  alias OMG.ChildChain.FeeServer
+  alias OMG.Eth
+  alias OMG.TestHelper
+
+  @moduletag :integration
+  @moduletag :child_chain
+
+  @eth Eth.zero_address()
+  @eth_hex Eth.Encoding.to_hex(@eth)
+  @fees %{@eth_hex => 0}
+
+  setup do
+    ignore_option = Application.fetch_env!(:omg_child_chain, :ignore_fees)
+    old_file_name = Application.fetch_env!(:omg_child_chain, :fee_specs_file_name)
+
+    {:ok, file_path, file_name} = TestHelper.write_fee_file(@fees)
+    Application.put_env(:omg_child_chain, :fee_specs_file_name, file_name)
+
+    on_exit(fn ->
+      File.rm(file_path)
+      Application.put_env(:omg_child_chain, :ignore_fees, ignore_option)
+      Application.put_env(:omg_child_chain, :fee_specs_file_name, old_file_name)
+
+      # without waiting tests interfere on :ets
```

Comment:
where's `:ets` used?

Reply:
In fee server - I seriously thinking to keep fee-map in FeeServer's state

Reply:
ah yes, you're right and moving it to process state makes sense as well!